File: Web/RMA-Web-Shell/projects/clientcare/src/app/policy-manager/shared/entities/role-player-policy-transaction-detail.ts
import { CategoryInsuredEnum } from "../enums/categoryInsuredEnum";

export class RolePlayerPolicyTransactionDetail {
  rolePlayerPolicyTransactionDetailId: number;
  rolePlayerPolicyTransactionId: number;
  productOptionId: number;
  
  categoryInsured: CategoryInsuredEnum
  rate: number;
  numberOfEmployees: number;
  totalEarnings: number;
  originalPremium: number;
  premium: number;
  liveInAllowance: number;
  effectiveFrom: Date;
  effectiveTo: Date;

  isDeleted: boolean;
  createdBy: string;
  createdDate: Date;
  modifiedBy: string;
  modifiedDate: Date;
}

File: Service/ClientCare/RMA.Service.ClientCare.Contracts/Entities/Member/RolePlayerPolicyTransactionDetail.cs
using RMA.Service.Admin.MasterDataManager.Contracts.Enums;

namespace RMA.Service.ClientCare.Contracts.Entities.Member
{
    public class RolePlayerPolicyTransactionDetail
    {
        public int RolePlayerPolicyTransactionDetailId { get; set; } // RolePlayerPolicyTransactionDetailId (Primary key)
        public int RolePlayerPolicyTransactionId { get; set; } // RolePlayerPolicyTransactionId
        public int ProductOptionId { get; set; } // ProductOptionId (length: 10)
        public CategoryInsuredEnum CategoryInsured { get; set; } // CategoryInsuredId
        public decimal Rate { get; set; } // Rate
        public int NumberOfEmployees { get; set; } // NumberOfEmployees
        public decimal TotalEarnings { get; set; } // TotalEarnings
        public decimal? OriginalPremium { get; set; }
        public decimal Premium { get; set; } // Premium
        public int LiveInAllowance { get; set; } // LiveInAllowance
        public System.DateTime EffectiveFrom { get; set; } // EffectiveFrom
        public System.DateTime EffectiveTo { get; set; } // EffectiveTo
        public bool IsDeleted { get; set; } // IsDeleted
        public string ModifiedBy { get; set; } // ModifiedBy (length: 50)
        public System.DateTime ModifiedDate { get; set; } // ModifiedDate
        public string CreatedBy { get; set; } // CreatedBy (length: 50)
        public System.DateTime CreatedDate { get; set; } // CreatedDate
    }
}